pallas-crypto/src/vrf/mod.rs

```diff
@@ -0,0 +1,159 @@
+use crate::hash::Hash;
+use crate::memsec::Scrubbed;
+use thiserror::Error;
+use vrf_dalek::vrf03::{PublicKey03, SecretKey03, VrfProof03};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("VrfError {0}")]
+    VrfError(#[from] vrf_dalek::errors::VrfError),
+}
+
+pub const VRF_SEED_SIZE: usize = 32;
+pub const VRF_PROOF_SIZE: usize = 80;
+pub const VRF_PUBLIC_KEY_SIZE: usize = 32;
+pub const VRF_SECRET_KEY_SIZE: usize = 32;
+pub const VRF_PROOF_HASH_SIZE: usize = 64;
+
+// Wrapper for VRF secret key
+pub struct VrfSecretKey {
+    secret_key_03: SecretKey03,
+}
+
+// Wrapper for VRF public key
+pub struct VrfPublicKey {
+    public_key_03: PublicKey03,
+}
+
+// Wrapper for VRF proof
+pub struct VrfProof {
+    proof_03: VrfProof03,
+}
+
+// Create a VrfSecretKey from a slice
+impl From<&[u8; VRF_SECRET_KEY_SIZE]> for VrfSecretKey {
+    fn from(slice: &[u8; VRF_SECRET_KEY_SIZE]) -> Self {
+        VrfSecretKey {
+            secret_key_03: SecretKey03::from_bytes(slice),
+        }
+    }
+}
+
+// Create a VrfPublicKey from a slice
+impl From<&[u8; VRF_PUBLIC_KEY_SIZE]> for VrfPublicKey {
+    fn from(slice: &[u8; VRF_PUBLIC_KEY_SIZE]) -> Self {
+        VrfPublicKey {
+            public_key_03: PublicKey03::from_bytes(slice),
+        }
+    }
+}
+
+// Create a VrfProof from a slice
+impl From<&[u8; VRF_PROOF_SIZE]> for VrfProof {
+    fn from(slice: &[u8; VRF_PROOF_SIZE]) -> Self {
+        VrfProof {
+            proof_03: VrfProof03::from_bytes(slice).unwrap(),
+        }
+    }
+}
+
+// Create a VrfPublicKey from a VrfSecretKey
+impl From<&VrfSecretKey> for VrfPublicKey {
+    fn from(secret_key: &VrfSecretKey) -> Self {
+        VrfPublicKey {
+            public_key_03: PublicKey03::from(&secret_key.secret_key_03),
+        }
+    }
+}
+
+// Wipe the secret key
+impl Scrubbed for VrfSecretKey {
+    fn scrub(&mut self) {
+        self.secret_key_03.to_bytes().scrub();
```

Comment:
we aren't really wiping the secret key here:

https://github.com/input-output-hk/vrf/blob/a3185620b72e6a9647285941b961021186f16693/src/vrf03.rs#L32-L35

The `to_bytes` functions returns a copy of the byte array of the secret key. It's still in memory. I wonder if we can either change that upstream (better choice) or be cheeky (not great) here:

```suggestion
        let mut bytes: [u8; VRF_SECRET_KEY_SIZE] = std::men::transmute(&mut self.secret_key_03);
```

I think the upstream change would make more sense but that's also a direction that can be taken.

Reply:
https://github.com/input-output-hk/vrf/pull/3

Reply:
upstream fix is merged.